src/main/java/qna/service/QnaService.java

```diff
@@ -6,50 +6,34 @@
 import org.springframework.transaction.annotation.Transactional;
 import qna.CannotDeleteException;
 import qna.NotFoundException;
-import qna.domain.*;
-
-import java.time.LocalDateTime;
-import java.util.ArrayList;
-import java.util.List;
+import qna.domain.Question;
+import qna.domain.QuestionRepository;
+import qna.domain.User;
 
 @Service
 public class QnaService {
+
     private static final Logger log = LoggerFactory.getLogger(QnaService.class);
 
     private QuestionRepository questionRepository;
     private DeleteHistoryService deleteHistoryService;
 
-    public QnaService(QuestionRepository questionRepository, DeleteHistoryService deleteHistoryService) {
+    public QnaService(QuestionRepository questionRepository,
+        DeleteHistoryService deleteHistoryService) {
         this.questionRepository = questionRepository;
         this.deleteHistoryService = deleteHistoryService;
     }
 
     @Transactional(readOnly = true)
     public Question findQuestionById(Long id) {
         return questionRepository.findByIdAndDeletedFalse(id)
-                .orElseThrow(NotFoundException::new);
+            .orElseThrow(NotFoundException::new);
     }
 
     @Transactional
     public void deleteQuestion(User loginUser, Long questionId) throws CannotDeleteException {
-        Question question = findQuestionById(questionId);
-        if (!question.isOwner(loginUser)) {
-            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
-        }
-
-        for (Answer answer : question.answers()) {
-            if (!answer.isOwner(loginUser)) {
-                throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
-            }
-        }
-
-        List<DeleteHistory> deleteHistories = new ArrayList<>();
-        question.setDeleted(true);
-        deleteHistories.add(new DeleteHistory(ContentType.QUESTION, questionId, question.getWriter(), LocalDateTime.now()));
-        for (Answer answer : question.answers()) {
-            answer.setDeleted(true);
-            deleteHistories.add(new DeleteHistory(ContentType.ANSWER, answer.getId(), answer.getWriter(), LocalDateTime.now()));
-        }
-        deleteHistoryService.saveAll(deleteHistories);
+        deleteHistoryService.saveAll(
```

Comment:
```suggestion
        final Question question = findQuestionById(questionId);
        final List<DeleteHistory> deleteHistories = question.delete(loginUser);
        deleteHistoryService.saveAll(deleteHistories);
```

변수로 나눠서 파라미터로 전달하는게 코드 파악하기에 더 좋지 않을까요?

Reply:
이 부분은 생각이 조금 다른 것 같습니다.
메모리를 사용하고 가비지 컬렉터의 대상이 될 수 있는 불필요한 변수를 생성할 필요는 없다고 생각합니다.
만약 구분하는 것이 목적이라면 최대한 메소드로 나누어 주어야 한다고 생각하는데 어떻게 생각하시나요?

Reply:
성능 이슈가 있는 게 아닌데 가비지 컬렉터를 고민해야되는 코드로 보이진 않네요.
저 변수를 추가해서 서비스에 이슈 성능에 생기는 걸까요?
읽기 좋은 코드를 작성하는게 더 중요한 경우도 있습니다.

Reply:
그리고 메서드로 구분하게 되면 왜 이 메서드를 추가햇는지 파악해야되니 일일이 메서드로 감쌀 필요는 없어보입니다.

Reply:
무조건 성능이 좋아지니까 메소드로 써야된다라는 것은 아니었습니다.
⌜이펙티브 자바⌟에서 지역변수 범위(scope)는 최소화해야 된다고 하며
⌜엘레강트 오브젝트⌟ 내용을 참고하면 빌더 메소드는 명사로 정의하라고 합니다.
변수의 수명을 짧게 유지하고 가독성을 위한다면 메소드를 이용하여 표현하는 것이 바람직하다고 생각했습니다.